state-synchronizer/src/state_synchronizer.rs

```diff
@@ -36,13 +35,72 @@ use tokio::{
     time::timeout,
 };
 
+/// The state distinguishes between the following fields:
+/// * highest_local_li is keeping the latest certified ledger info
+/// * synced_trees is keeping the latest state in the transaction accumulator and state tree.
+///
+/// While `highest_local_li` can be used for helping the others (corresponding to the highest
+/// version we have a proof for), `synced_trees` is used for retrieving missing chunks
+/// for the local storage.
+#[derive(Clone)]
+pub struct SynchronizationState {
+    highest_local_li: LedgerInfoWithSignatures,
+    synced_trees: ExecutedTrees,
+    // Corresponds to the current epoch if the highest local LI is in the middle of the epoch,
+    // or the next epoch if the highest local LI is the final LI in the current epoch.
+    trusted_epoch: EpochState,
+}
+
+impl SynchronizationState {
+    pub fn new(
+        highest_local_li: LedgerInfoWithSignatures,
+        synced_trees: ExecutedTrees,
+        current_epoch_state: EpochState,
+    ) -> Self {
+        let trusted_epoch = highest_local_li
+            .ledger_info()
+            .next_epoch_state()
+            .cloned()
+            .unwrap_or(current_epoch_state);
+        SynchronizationState {
+            highest_local_li,
+            synced_trees,
+            trusted_epoch,
+        }
+    }
+
+    pub fn committed_epoch(&self) -> u64 {
+        self.highest_local_li.ledger_info().epoch()
+    }
+
+    pub fn committed_ledger_info(&self) -> LedgerInfoWithSignatures {
+        self.highest_local_li.clone()
+    }
+
+    pub fn committed_version(&self) -> u64 {
+        self.highest_local_li.ledger_info().version()
+    }
+
+    pub fn epoch(&self) -> u64 {
+        self.trusted_epoch.epoch
+    }
+
+    /// The highest available version in the local storage (even if it's not covered by the LI).
+    pub fn highest_version_in_local_storage(&self) -> u64 {
```

Comment:
if we want to commit to the terminology used historically, we should really do that in both directions...

committed -- what we can produce arbitrary proofs of correctness for or more formally the highest version for which we have all ledger state and a ledger info
synced -- what we have viewed as proven but lack sufficient information to produce arbitrary proofs or more formally we lack some ledger state for the next highest li version to be viable to produce arbitrary proofs (e.g., next li is n + 10 and we only have transactions through n+8)

we should probably write that somewhere and with better phrasing if you can think of it.

then this value isn't highest version in local storage but rather synced_version

Reply:
Aah.. I agree with you! I actually have another PR ready to send across that does exactly this. It also does a bunch of other stuff -- will defer to that PR.